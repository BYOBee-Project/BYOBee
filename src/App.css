@import url('https://fonts.googleapis.com/css2?family=Comfortaa:wght@300;700&family=Londrina+Solid&display=swap');

/* font-family: 'Comfortaa', cursive;
font-family: 'Londrina Solid', cursive; */

:root {
  --cg-blue: #0081a7ff;
  --verdigris: #00afb9ff;
  --light-yellow: #fdfcdcff;
  --peach-puff: #fed9b7ff;
  --bittersweet: #f07167ff;
}

body {
  margin: 0;
  background-color: rgba(254, 217, 183, 0.233);
}

.app {
  margin: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

a:link {
  text-decoration: none;
  color: black;
}

a:visited {
  color: black;
}

button {
  border: none;
  background-color: var(--cg-blue);
  font-size: 2vh;
  padding: 1vh;
  color: white;
  border-radius: 5px;
  font-family: 'Londrina Solid', cursive;
}

button:hover {
  transform: scale(1.1);
}

button a {
  color: white;
}

button a:visited {
  color: white;
}

.loader {
  font-family: 'Londrina Solid', cursive;
  margin: 10vh;
}
