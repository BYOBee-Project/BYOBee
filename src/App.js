import './App.css';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Home from './views/Home/Home';
import BeeDetail from './components/BeeDetail/BeeDetail';
import Header from './components/Header/Header';
import About from './components/About/About';
import Resources from './components/Resources/Resources';
import Auth from './views/Auth/Auth';
import Profile from './views/Profile/Profile';
import { useState } from 'react';
import { getUser } from './services/users';
import BeeForm from './components/BeeForm/BeeForm';

function App() {
  const [currentUser, setCurrentUser] = useState(getUser());

  return (
    <div className="app">
      <BrowserRouter>
        <Header currentUser={currentUser} />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/bees" component={Home} />
          <Route exact path="/bees/:id" component={BeeDetail} />
          <Route exact path="/about" component={About} />
          <Route exact path="/resources" component={Resources} />
          <Route exact path="/profile">
            <Profile setCurrentUser={setCurrentUser} currentUser={currentUser} />
          </Route>
          <Route exact path="/auth">
            <Auth setCurrentUser={setCurrentUser} />
          </Route>
          <Route exact path="/beeform/:id">
            <BeeForm currentUser={currentUser} />
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
