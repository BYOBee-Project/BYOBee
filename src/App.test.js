import { render, screen } from '@testing-library/react';
import App from './App';

test('App should render home page', async () => {
  render();
  const container = render(<App />);
  await screen.findByText('Get to know the bees in your backyard!');
  expect(container).toMatchSnapshot();
});
