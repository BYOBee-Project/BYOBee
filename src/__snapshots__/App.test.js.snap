// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="app"
      >
        <div
          class="header"
        >
          <header>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
            <a
              href="/about"
            >
              About
            </a>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <img
                class="logo"
                src="black-logo.png"
              />
            </a>
            <a
              href="/resources"
            >
              Resources
            </a>
            <a
              href="/auth"
            >
              Sign-In
            </a>
          </header>
        </div>
        <div
          class="home"
        >
          <div
            class="home-intro"
          >
            <h2
              class="home-title"
            >
              Get to know the bees in your backyard!
            </h2>
            <p
              class="home-info"
            >
              Did you know there are over 500 bee species in Oregon? Use this guide to identify the bees you see, and then build your virtual collection by reporting those encounters. Enjoy connecting with the natural world while providing valuable data to pollinator conservation efforts! Click a bee to get started.
            </p>
          </div>
          <div
            class="bee-list"
          >
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Alfalfa Leafcutter Bee
              </h3>
              <a
                href="/bees/6"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-rotundata/megachile-rotundata-1.jpg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Black-Notched Bumblebee
              </h3>
              <a
                href="/bees/3"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/bombus-bifarius/bombus-bifarius-1.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Blue Orchard Mason Bee
              </h3>
              <a
                href="/bees/2"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/osmia-lignaria/osmia-lignaria-1.jpg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Brown-Belted Bumblebee
              </h3>
              <a
                href="/bees/7"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/bombus-griseocollis/Bombus-griseocollis-2.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Honey Bee (Drone)
              </h3>
              <a
                href="/bees/12"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/apis-mellifera-drone/apis-mellifera-drone-1.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Honey Bee (Queen)
              </h3>
              <a
                href="/bees/10"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/apis-mellifera-queen/apis-mellifera-queen-1.jpg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Honey Bee (Worker)
              </h3>
              <a
                href="/bees/9"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/apis-mellifera-worker/apis-mellifera-worker-1.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Long-Horned Bee
              </h3>
              <a
                href="/bees/11"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/melissodes-spp/melissodes-spp-1.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Prunus Miner Bee
              </h3>
              <a
                href="/bees/1"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/andrena-prunorum/andrena-prunorum-1.jpg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Pugnacious Leafcutter Bee
              </h3>
              <a
                href="/bees/8"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-pugnata/megachile-pugnata-1.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Small Carpenter Bee
              </h3>
              <a
                href="/bees/4"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/ceratina-spp/ceratina-spp-1.jpeg"
                />
              </a>
            </div>
            <div
              class="bee-card"
            >
              <h3
                class="bee-card-title"
              >
                Yellow-Faced Bumblebee
              </h3>
              <a
                href="/bees/5"
              >
                <img
                  class="card-image"
                  src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/bombus-vosnesenskii/bombus-vosnesenskii-1.jpg"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <div
        class="header"
      >
        <header>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
          <a
            href="/about"
          >
            About
          </a>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              class="logo"
              src="black-logo.png"
            />
          </a>
          <a
            href="/resources"
          >
            Resources
          </a>
          <a
            href="/auth"
          >
            Sign-In
          </a>
        </header>
      </div>
      <div
        class="home"
      >
        <div
          class="home-intro"
        >
          <h2
            class="home-title"
          >
            Get to know the bees in your backyard!
          </h2>
          <p
            class="home-info"
          >
            Did you know there are over 500 bee species in Oregon? Use this guide to identify the bees you see, and then build your virtual collection by reporting those encounters. Enjoy connecting with the natural world while providing valuable data to pollinator conservation efforts! Click a bee to get started.
          </p>
        </div>
        <div
          class="bee-list"
        >
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Alfalfa Leafcutter Bee
            </h3>
            <a
              href="/bees/6"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-rotundata/megachile-rotundata-1.jpg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Black-Notched Bumblebee
            </h3>
            <a
              href="/bees/3"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/bombus-bifarius/bombus-bifarius-1.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Blue Orchard Mason Bee
            </h3>
            <a
              href="/bees/2"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/osmia-lignaria/osmia-lignaria-1.jpg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Brown-Belted Bumblebee
            </h3>
            <a
              href="/bees/7"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/bombus-griseocollis/Bombus-griseocollis-2.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Honey Bee (Drone)
            </h3>
            <a
              href="/bees/12"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/apis-mellifera-drone/apis-mellifera-drone-1.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Honey Bee (Queen)
            </h3>
            <a
              href="/bees/10"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/apis-mellifera-queen/apis-mellifera-queen-1.jpg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Honey Bee (Worker)
            </h3>
            <a
              href="/bees/9"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/apis-mellifera-worker/apis-mellifera-worker-1.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Long-Horned Bee
            </h3>
            <a
              href="/bees/11"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/melissodes-spp/melissodes-spp-1.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Prunus Miner Bee
            </h3>
            <a
              href="/bees/1"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/andrena-prunorum/andrena-prunorum-1.jpg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Pugnacious Leafcutter Bee
            </h3>
            <a
              href="/bees/8"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/megachile-pugnata/megachile-pugnata-1.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Small Carpenter Bee
            </h3>
            <a
              href="/bees/4"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/ceratina-spp/ceratina-spp-1.jpeg"
              />
            </a>
          </div>
          <div
            class="bee-card"
          >
            <h3
              class="bee-card-title"
            >
              Yellow-Faced Bumblebee
            </h3>
            <a
              href="/bees/5"
            >
              <img
                class="card-image"
                src="https://purcoqerkuxhmrkzgmyk.supabase.in/storage/v1/object/public/images/bees/bombus-vosnesenskii/bombus-vosnesenskii-1.jpg"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
